connectors/sources/confluence.py

```diff
@@ -0,0 +1,247 @@
+        if self.session is None:
+            self._generate_session()
+        if self.ssl_enabled and (self.certificate == "" or self.certificate is None):
+            self.ssl_ctx = ssl_context(certificate=self.certificate)
+
+    async def validate_config(self):
+        """Validates whether user input is empty or not for configuration fields
+
+        Raises:
+            Exception: Configured fields can't be empty.
+            Exception: SSL certificate must be configured.
+            Exception: Concurrent downloads can't be set more than maximum allowed value.
+        """
+        logger.info("Validating Confluence Configuration...")
+
+        connection_fields = (
+            ["host_url", "service_account_id", "api_token"]
+            if self.is_cloud
+            else ["host_url", "username", "password"]
+        )
+        default_config = self.get_default_configuration()
+
+        if empty_connection_fields := [
+            default_config[field]["label"]
+            for field in connection_fields
+            if self.configuration[field] == ""
+        ]:
+            raise Exception(
+                f"Configured keys: {empty_connection_fields} can't be empty."
+            )
+        if self.ssl_enabled and self.certificate == "":
+            raise Exception("SSL certificate must be configured.")
+
+        if self.concurrent_downloads > MAX_CONCURRENT_DOWNLOADS:
+            raise Exception(
+                f"Configured concurrent downloads can't be set more than {MAX_CONCURRENT_DOWNLOADS}."
+            )
+
+    async def _api_call(self, url):
+        """Make a GET call for Atlassian API using the passed url with retry for the failed API calls.
+
+        Args:
+            url: Request URL to hit the get call
+
+        Raises:
+            exception: An instance of an exception class.
+
+        Yields:
+            response: Client response
+        """
+        retry_counter = 0
+        while True:
+            try:
+                async with self.session.get(
+                    url=url,
+                    ssl=self.ssl_ctx,
+                ) as response:
+                    yield response
+                    break
+            except Exception as exception:
+                if isinstance(
+                    exception,
+                    ServerDisconnectedError,
+                ):
+                    await self.session.close()  # pyright: ignore
+                    self._generate_session()
+                retry_counter += 1
+                if retry_counter > self.retry_count:
+                    raise exception
+                logger.warning(
+                    f"Retry count: {retry_counter} out of {self.retry_count}. Exception: {exception}"
+                )
+                await self._sleeps.sleep(RETRY_INTERVAL**retry_counter)
+
+    async def ping(self):
+        """Verify the connection with Confluence"""
+        self.initialize_config()
```

Comment:
Why do we call `initialize_config` here? I think we should move that probably to the constructor so that `ping` doesn't have the initialization side effect.

Reply:
Updated with commit id: [refactor initialize_config](https://github.com/elastic/connectors-python/pull/568/commits/b4304823b38f6f5553469c2435c15f52611495ce)
[b430482](https://github.com/elastic/connectors-python/pull/568/commits/b4304823b38f6f5553469c2435c15f52611495ce)